templates/todo/api/java/src/main/java/com/microsoft/azure/simpletodo/configuration/WebConfiguration.java

```diff
@@ -22,27 +29,42 @@ public WebMvcConfigurer webConfigurer() {
         return new WebMvcConfigurer() {
             @Override
             public void addCorsMappings(CorsRegistry registry) {
-                String apiUrl = System.getenv("REACT_APP_WEB_BASE_URL");
-
-                if (apiUrl != null) {
-                    String localHostUrl = "http://localhost:3000/";
-                    registry.addMapping("/**").allowedOrigins("https://portal.azure.com",
-                            "https://ms.portal.azure.com", localHostUrl,
-                            apiUrl).allowedMethods("*").allowedHeaders("*");
-                    String fileName = Thread.currentThread().getStackTrace()[1].getFileName();
-                    File file = new File(fileName);
-                    String absolutePath = file.getAbsolutePath();
-                    System.out.println(
-                            "CORS with " + localHostUrl
-                                    + " is allowed for local host debugging. If you want to change pin number, go to "
-                                    + absolutePath);
+                // env.ENABLE_ORYX_BUILD is only set on Azure environment during azd provision for todo-templates
+                // You can update this to env.JAVA_ENV if your app is using `development` to run locally and another value
+                // when the app is running on Azure (like production or stating)
+                String runningOnAzure = System.getenv("ENABLE_ORYX_BUILD");
+
+                if (runningOnAzure != null) {
+                    ArrayList<String> origins = new ArrayList<>();
+                    origins.add("https://portal.azure.com");
+                    origins.add("https://ms.portal.azure.com");
 
+                    if (localhostOrigin != "") {
+                        origins.add(localhostOrigin);
+                        String fileName = Thread.currentThread().getStackTrace()[1].getFileName();
+                        File file = new File(fileName);
+                        String absolutePath = file.getAbsolutePath();
+                        System.out.println(
+                            "Allowing requests from" + localhostOrigin + ". To change or disable, go to " + absolutePath
+                        );
+                    }
```

Comment:
What is your take on how you would implement this if you were writing an application outside of `azd`? Is there a different approach here? I've begun to more and more dislike how our templates are written in a sandboxed world that doesn't work out in practice, I wonder if there's a way to do something different here.

Reply:
It is usually idiomatic...  for nodejs, for example, it is common to use env.NODE_ENV to define where is your app running and set the different configuration/behavior based on that.

On java (bootspring) there is a concept of environment configuration where you can override settings using conifg files.

I am not sure about how python handles this.

But, for the todo templates, we are not taking idiomatic decisions for the app and its infrastructure.  We are using the same infrastructure and trying to adapt each language to work within that infra.

My preference would be to have specific infrastructure configuration per language, instead of per service-target.  But that would bring more branches to maintain within Azure-dev repo.

My second preference would be to pull the todo templates our of the Azure-Dev repo and maintain them individually, where each one can define their own idiomatic requirements.

Reply:
@vhvb1989 I think my high-level take is that code that depends on a magic "ENV_NAME" is less optimal if I was designing a service. I would much rather have it be an environment specific configuration that is passed.

If I want my something to trust my localhost environment, I should set a configuration for "TRUSTED_HOSTS" to include my localhost URL?

Reply:
I'm also not opposed to the env name check (depends on use-case and impl). That does bring it better than what we were doing before, which is checking for very specific Azure environment variables specifically.